EDF.py

```diff
@@ -8,32 +41,44 @@
 import re
 import warnings
 
+
 def padtrim(buf, num):
     num -= len(buf)
-    if num>=0:
-        # pad the input to the specified length
-        return str(buf) + ' ' * num
-    else:
-        # trim the input to the specified length
-        return buf[0:num]
-
-####################################################################################################
-# the EDF header is represented as a tuple of (meas_info, chan_info)
-# meas_info should have ['record_length', 'magic', 'hour', 'subject_id', 'recording_id', 'n_records', 'month', 'subtype', 'second', 'nchan', 'data_size', 'data_offset', 'lowpass', 'year', 'highpass', 'day', 'minute']
-# chan_info should have ['physical_min', 'transducers', 'physical_max', 'digital_max', 'ch_names', 'n_samps', 'units', 'digital_min']
-####################################################################################################
+    return (str(buf) + ' ' * num) if (num >= 0) else buf[0:num]
```

Comment:
This is just a short-hand expression of the multi-line if-else condition. The explanation of the code is written in the documentation of the function, which you earlier advised to put in separate pull request. Hence it was removed here. You can open the earlier commit and compare the code with the documentation if required.
![fun_padtrim_doc](https://user-images.githubusercontent.com/42271103/87564634-42901680-c6de-11ea-9ba2-afd1b57a288e.JPG)

I expected these confusions to happen. I should have explained earlier why it would be better to consider both the documentation and code edits to review side by side. This is my bad. Let me know if I can do anything to alleviate this.

Reply:
I  understand the Python syntax. I don't understand the need to change to a different syntax. Changes just to make changes aren't necessary.

Reply:
> Do you not think that this is somewhat better than a multi-line if-else?

No, I don't. If for no other reason than the order is inverted compared to most other programming constructs (the test occurs between values, which is weird in programming languages, see e.g. the tertiary operator in C/C++/Java ` test ? val_if_true : val_if_false` or R's `ifelse(test, val_if_true, val_if_false`.

> The backstory for this is that when I was making these formatting I was working with embedded engineers who are too paranoid about saving space. I picked up the habit because I had to write similar codes.

The length of the source does not matter. The length of the compiled code does. Source code is for humans. Parsed/compiled code is for computers.

Reply:
> the order is inverted compared to most other programming constructs
Yes, that is just the philosophy of python . It's meant to be simpler like `val_if_true if test is true else val_if_false`. Though the syntax is not a ternary operator. It is simply an one-line if-else expression such as lambda function expressions, added back in Python 2.5.

https://mail.python.org/pipermail/python-dev/2005-September/056846.html

Reply:
Again, I'm very familiar with both the syntax and its history in Python. (The same holds for all of my other points: I don't require an introduction to Python, nor an explanation of the differences between Python 2 and Python 3.) Note that I did not call it a tertiary operator, but compared it the tertiary operator in other languages.

The philosophy of Python not to invert order. In the Zen of Python, it's quite clear that there should be a preferred obvious way to do something. The multiline `if` statement is obvious and doesn't involve an unnecessary change. Notice in the link you provide that the former BDFL even mentions a "long discussion" for adding the single-line `if` expression. The long discussion was because simply adding that expression into the language potentially violated the Zen:
> Readability counts. 
....
> There should be one-- and preferably only one --obvious way to do it.

There are two ways here in Python, but the multiline statement is the older and more obvious route.

I'll concede that other programmers find the one-liner easy/ier to read. But the multiline version is not hard to read *and it doesn't involve a cosmetic change to somebody else's codebase*.

Reply:
I feel we are having a developer's feud about a line of code. 😅 I do not wish to do this, and I understand your point, and your reasoning is correct. Yet I, for some reason, feel rigid about this line. 😂 

Though what you pointed out is correct. I do not want to be remembered as the guy doing _cosmetic changes_ to this codebase and then being like "I shall defend this short braid of hair till death!". That's not me, so I am making the change now. 💔